### DISCUSSION TITLE
default value of a column is reset to null on an update mutation?

### DISCUSSION THREAD

**Discussion by stayingcool** [hasura/graphql-engine#6283] [General]
Hello Folks, 

Do you know why the default value of a column is reset to null on an update mutation? 

Column Definition is as follows, When the record is inserted without the completedUserID, then completedUserID should be having a fixed value of "None". On insertion, I can see a default value of "None".

> completedUserID - text, nullable, default: 'None'::text

However, when the update mutation is invoked to update some other fields, completedUserID is reset to null! 

Any clues as to what's going on?

**Comment by praveenweb** [comment]
@stayingcool - Assuming it has a foreign key constraint, Is the completedUserID having an `On update violation` set to `null`? That could be a cause of reset of values to null.

**Reply by stayingcool** [reply]
@praveenweb - There are no foreign key constraints. The table has no relationship with other tables.

I created a simple table (items) with these 3 columns:
**Columns**
**id** - integer, primary key, unique, default: nextval('items_id_seq'::regclass)
**name** - text
**category** - text, nullable, default: 'none'::text
 
------
**Mutation (Insert an item) -** Works Fine

```
mutation AddItem($name: String!, $category: String) {
  insert_items_one(object: {name: $name, , category: $category}) {
    category
    id
    name
  }
}
```
**Variable**
```
{
  "name": "Coffee",
  "category": "Beverage"
}
```
**Response**
```
{
  "data": {
    "insert_items_one": {
      "category": "Beverage",
      "id": 6,
      "name": "Coffee"
    }
  }
}
```
------
 **Mutation (Update an item by pk) -** Not expecting the category column to be set to null

```
mutation UpdateItem($id: Int!, $name: String!, $category: String) {
  update_items_by_pk(pk_columns: {id: $id}, _set: {name: $name, category: $category}) {
    category
    id
    name
  }
}
```
**Variable**
```
{
   "id":6,
  "name": "Cold Coffee"
}
```
**Response**
```
{
  "data": {
    "update_items_by_pk": {
      "category": null,
      "id": 6,
      "name": "Cold Coffee"
    }
  }
}
```

If the category has been made optional in the mutation definition then why is it set to null after invocation?

**Reply by cdhaadmin** [reply]
@stayingcool 
I think this is a bug. For some reason the graphql-engine is sending NULL as the category, instead of not sending anything for that field. It doesn't happen if that variable is not in the mutation.

I would try in the latest version v1.3.4.  There were some changes in the query builder and how the variables are populated.  I know that in the `where` if the value does not exist in the variables it will ignore, before it wouldn't.  But I don't know if that will happen in an input type.  If the issue still happens in the latest version I would open a Github Issue.

You probably know this already but as a work around you can pass the default value in the mutation.

```javascript
mutation UpdateItem($id: Int!, $name: String!, $category: String = "none") {
  update_items_by_pk(pk_columns: {id: $id}, _set: {name: $name, category: $category}) {
    category
    id
    name
  }
}
```
It's not just in the update that is happening.  It will happen on an insert as well.

**Reply by stayingcool** [reply]
@cdhaadmin - Thanks for your valuable feedback. Sure thanks for the work-around of passing the default value. I'm just concerned about updating any field unless it is for a valid transaction.

By the way, I'm right on Hasura Cloud (v1.3.3-cloud.2). I'm under the assumption that the updates should happen automatically? 

@praveenweb - Who can confirm that this is a defect or not from Hasura Team? Is there a way to convert this to a defect if it is indeed a defect?
